lib/cli-engine.js

```diff
@@ -116,6 +116,66 @@ function calculateStatsPerRun(results) {
 }
 
 /**
+ * Performs multiple autofix passes over the text until as many fixes as possible
+ * have been applied.
+ * @param {string} text The source text to apply fixes to.
+ * @param {Object} config The ESLint config object to use.
+ * @param {Object} options The ESLint options object to use.
+ * @param {string} options.filename The filename from which the text was read.
+ * @param {boolean} options.allowInlineConfig Flag indicating if inline comments
+ *      should be allowed.
+ * @returns {Object} The result of the fix operation as returned from the
+ *      SourceCodeFixer.
+ * @private
+ */
+function multipassFix(text, config, options) {
+
+    var messages = [],
+        fixedResult,
+        fixed = false,
+        passNumber = 0,
+        lastMessageCount;
+
+    /**
+     * This loop continues until one of the following is true:
+     *
+     * 1. No more fixes have been applied.
+     * 2. There are no more linting errors reported.
+     * 3. The number of linting errors is no different between two passes.
+     *
+     * That means anytime a fix is successfully applied, there will be another pass.
+     * Essentially, guaranteeing a minimum of two passes.
+     */
+    do {
+        passNumber++;
+        lastMessageCount = messages.length;
+
+        debug("Linting code for " + options.filename + " (pass " + passNumber + ")");
+        messages = eslint.verify(text, config, options);
+
+        debug("Generating fixed text for " + options.filename + " (pass " + passNumber + ")");
+        fixedResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+
+        // keep track if any fixes were ever applied - important for return value
+        fixed = fixed || fixedResult.fixed;
+
+        // update to use the fixed output instead of the original text
+        text = fixedResult.output;
+
+    } while (
+        fixedResult.fixed && fixedResult.messages.length > 0 &&
+        fixedResult.messages.length !== lastMessageCount
```

Comment:
I guess this condition can enter into an infinite loop if the count repeats `+1` and `-1`.
I think good that it continues until the minimum value of the count was not decreased while N times.
For example:

```js
var minMessageCount = Number.POSITIVE_INFINITY;
var notDecreasingCount = 0;
do {
    if (messages.length < minMessageCount) {
        minMessageCount = messages.length;
        notDecreasingCount = 0;
    } else {
        notDecreasingCount += 1;
    }

    //....
} while (fixedResult.fixed && fixedResult.messages.length > 0 && notDecreasingCount < 4);
```